dio/lib/src/multipart_file.dart

```diff
@@ -140,4 +140,14 @@ class MultipartFile {
     _isFinalized = true;
     return _stream;
   }
+
+  /// Restore MultipartFile, returning a new instance of the same object. This is useful if your request failed and you wish
+  /// to sistematically retry it, for example a 401 error that can be solved by refreshing the token.
+  MultipartFile restoreMultipartFile() => MultipartFile(
```

Comment:
I'm not sure `restore` is a good word here because it represents manipulating the original object but here we produce a new one. Any other ideas? Other words like `copy` or `duplicate` might fit more in this case.

Reply:
humm, how about `reset`?

Reply:
I renamed it as duplicateMultipartFile, but let me know if you think reset would fit better :)